### PR TITLE
VM->Broker conversion for api/test_errata.py

### PR DESCRIPTION
**Pull request by JacobCallahan**
Converted api/test_errata.py
Also included some new fixtures and enhancements to the ContentHost
class
RHEL6 and RHEL8 tests can't be run due to needed template enhancements.

All non-rhel6/8 tests pass locally at the same percentage as before.

### CODE REVIEW

File: pytest_fixtures/xdist.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import random
 
 import pytest
-from broker.broker import VMBroker
+from broker import VMBroker
```
Comment (jameerpathan111) [inline pytest_fixtures/xdist.py:7]: Is this an intended change?

File: tests/foreman/api/test_errata.py [hunk 2]
```diff
@@ -199,25 +179,23 @@ def test_positive_bulk_install_package(module_org, activation_key, custom_repo,
...
-            host_ids=[host_id],
-            package_name=FAKE_1_CUSTOM_PACKAGE_NAME,
-            via_ssh=False,
-            rpm_package_name=FAKE_2_CUSTOM_PACKAGE,
-        )
+    rhel7_contenthost.install_katello_ca()
+    rhel7_contenthost.register_contenthost(module_org.label, activation_key.name)
+    assert rhel7_contenthost.subscribed
+    rhel7_contenthost.enable_repo(constants.REPOS['rhst7']['id'])
+    rhel7_contenthost.install_katello_agent()
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_errata.py:186]: Can you implement a module-level fixture just to optimize these repeated steps? rather than using plain `rhel7_contenthost`, also for rhel6/8 clients if necessary.

File: tests/foreman/api/test_errata.py [hunk 3]
```diff
@@ -199,25 +179,23 @@ def test_positive_bulk_install_package(module_org, activation_key, custom_repo,
...
+    host_id = entities.Host().search(query={'search': f'name={rhel7_contenthost.hostname}'})[0].id
```
Comment (jameerpathan111) [inline tests/foreman/api/test_errata.py:187]: @JacobCallahan We can use `nailgun_host` here?

File: robottelo/hosts.py [hunk 4]
```diff
@@ -52,6 +53,23 @@ class ContentHostError(Exception):
 
 class ContentHost(Host):
     run = Host.execute
+    subscribed = False
+
+    @cached_property
+    def nailgun_host(self):
+        """If this host is subscribed, provide access the it's nailgun object"""
```
Comment (swadeley) [inline robottelo/hosts.py:60]:
```suggestion
        """If this host is subscribed, provide access to it's nailgun object"""
```

**Review by swadeley — APPROVED** [review]
ACK

File: tests/foreman/api/test_errata.py (hunk 3 above)
Comment (swadeley) [reply tests/foreman/api/test_errata.py:187]: I see `from nailgun import entities`, so I think `Host()` is from Nailgun.  I tried to grep for `nailgun_host` but could not find it.

File: robottelo/hosts.py [hunk 5]
```diff
@@ -52,6 +53,23 @@ class ContentHostError(Exception):
...
+        if self.subscribed:
+            return entities.Host().search(query={'search': self.hostname})[0]
+
+    @property
+    def subscribed(self):
+        """Boolean representation of a content host's subscription status"""
+        return 'Status: Unknown' not in self.execute('subscription-manager status').stdout
```
Comment (omkarkhatavkar) [inline robottelo/hosts.py:67]: Any specific reason for adding indirect approach that host is registered, you can also able to verify the status as `Status: Current` or `subscription-manager list | egrep -i 'status: '`

File: pytest_fixtures/broker.py [hunk 6]
```diff
@@ -67,10 +67,19 @@ def rhel7_contenthost_module():
 @pytest.fixture
 def rhel8_contenthost():
     """A fixture that provides a content host object based on the rhel8 nick"""
+    pytest.skip('Skipped until rhel8 templates have wget added')
```
Comment (tpapaioa) [inline pytest_fixtures/broker.py:70]: Why is `wget` necessary to use the fixture? You can provision and register a system without it.

File: pytest_fixtures/broker.py [hunk 7]
```diff
@@ -67,10 +67,19 @@ def rhel7_contenthost_module():
...
     with VMBroker(nick='rhel8', host_classes={'host': ContentHost}) as host:
         yield host
 
 
+@pytest.fixture
+def rhel6_contenthost():
+    """A function-level fixture that provides a content host object based on the rhel6 nick"""
+    pytest.skip('Waiting on RHEL 6 SatLab templates')
+    with VMBroker(nick='rhel6', host_classes={'host': ContentHost}) as host:
```
Comment (tpapaioa) [inline pytest_fixtures/broker.py:79]: Use `DISTRO_RHEL6`, `DISTRO_RHEL7`, and `DISTRO_RHEL8` from `robottelo.constants`.

File: robottelo/hosts.py (hunk 5 above)
Comment (tpapaioa) [reply robottelo/hosts.py:67]: There are multiple statuses possible, 

- fully subscribed: `Overall Status: Current` (return code 0)
- partially or fully unsubscribed: `Overall Status: Invalid` (return code 1)
- unregistered: `Overall Status: Unknown` (return code 1)

File: tests/foreman/api/test_errata.py [hunk 8]
```diff
@@ -20,49 +20,28 @@
...
-from robottelo.constants import REAL_0_ERRATA_ID
-from robottelo.constants import REAL_0_RH_PACKAGE
-from robottelo.constants import REAL_1_ERRATA_ID
-from robottelo.constants import REAL_2_ERRATA_ID
-from robottelo.constants import REPOS
-from robottelo.constants import REPOSET
-from robottelo.constants.repos import CUSTOM_SWID_TAG_REPO
-from robottelo.constants.repos import FAKE_3_YUM_REPO
-from robottelo.constants.repos import FAKE_9_YUM_REPO
+from robottelo.constants import repos
```
Comment (tpapaioa) [inline tests/foreman/api/test_errata.py:32]: What's the reason for this change?

**Review by tpapaioa — CHANGES_REQUESTED** [review]

File: pytest_fixtures/broker.py (hunk 6 above)
Comment (swadeley) [reply pytest_fixtures/broker.py:70]: @JacobCallahan is this because of `wget` in `robottelo/hosts.py `? If so, see:
https://github.com/SatelliteQE/robottelo/pull/8556#discussion_r625625501

File: tests/foreman/api/test_errata.py (hunk 2 above)
Comment (JacobCallahan) [reply tests/foreman/api/test_errata.py:186]: I'd prefer the tests to be more explicit in their behavior since it is more clear exactly what's going on.

File: tests/foreman/api/test_errata.py (hunk 3 above)
Comment (JacobCallahan) [reply tests/foreman/api/test_errata.py:187]: sure can! i'll update that in these places since that was a later change. good catch!

File: robottelo/hosts.py (hunk 4 above)
Comment (JacobCallahan) [reply robottelo/hosts.py:60]: ha! you're absolutely correct

File: robottelo/hosts.py (hunk 5 above)
Comment (JacobCallahan) [reply robottelo/hosts.py:67]: Tasos nailed it. The second status is very common since we also have tests where we only have custom subscriptions.

File: pytest_fixtures/broker.py (hunk 6 above)
Comment (JacobCallahan) [reply pytest_fixtures/broker.py:70]: @swadeley iirc there may be other places where it is needed, but after that change is introduced I can rebase and test again.

File: tests/foreman/api/test_errata.py (hunk 8 above)
Comment (JacobCallahan) [reply tests/foreman/api/test_errata.py:32]: IMO it looks better to have one import instead of several from the same location. it is also more explicit where those constants come from when used in code (more clear when something is a local/vs imported constant).

File: pytest_fixtures/broker.py (hunk 6 above)
Comment (JacobCallahan) [reply pytest_fixtures/broker.py:70]: yep. just confirmed that wget is used by helper methods elsewhere. i'm fine with us waiting on a rhel8 template that meets our requirements.

**Comment by JacobCallahan** [comment]
@jameerpathan111 @Gauravtalreja1 @tpapaioa @omkarkhatavkar I've updated the pr to address your comments

**Review by jameerpathan111 — APPROVED** [review]
Ack

File: robottelo/hosts.py [hunk 9]
```diff
@@ -52,6 +53,23 @@ class ContentHostError(Exception):
 
 class ContentHost(Host):
     run = Host.execute
+    subscribed = False
+
+    @cached_property
+    def nailgun_host(self):
+        """If this host is subscribed, provide access ot its nailgun object"""
+        if self.subscribed:
```
Comment (mshriver) [inline robottelo/hosts.py:61]: Since this depends on whether or not its subscribed, I wouldn't cache the property.

If its referenced before the host is subscribed, then it'll get cached to `None` and its on the caller to delete it before calling it again - making the caller have to check `host.subscribed` anyway.

**Review by mshriver — APPROVED** [review]
